### DISCUSSION TITLE
Banning on specific port with Shorewall

### DISCUSSION THREAD

**Discussion by robozb** [fail2ban/fail2ban#3286] [General]
Dear Members,

Is it possible: Banning on specific port with Shorewall?

Thanks: Bela

**Comment by sebres** [comment]
The fail2ban action from this repository cannot do that. Let alone I don't think it'd be even possible via [shorewall](https://www.systutorials.com/docs/linux/man/8-shorewall/) command line to restrict the destination port(s) without to create special zone or low-level table in net filter. If you know the syntax how it'd be possible you can simply create own action based on [action.d/shorewall.conf](https://github.com/fail2ban/fail2ban/blob/master/config/action.d/shorewall.conf) similar to another single or multi-ports actions of fail2ban. The tag `<port>` contains the single port or a port range and can be used in every action (on `actionban` called for every IP as well as on `actionstart` called once per jail to initialize the action).

But why one needs to use it at all? Shorewall is just a high-level tool for configuring low-level netfilter like iptables or nftables. It is rather something for humans not for automated tools.
Why fail2ban must use still one interim interface instead of invocation of kernel netfilter directly?

**Comment by robozb** [comment]
Dear Sebres, Thank you very much your lot of infos!
